File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/shell_script/ShellScriptConfigurationSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.shell_script;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.configurations.ConfigurationSettings;
import java.nio.file.Path;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableShellScriptConfigurationSettings.class)
public interface ShellScriptConfigurationSettings extends ConfigurationSettings {

  Path scriptPath();

  Optional<String> scriptOptions();

  Optional<InterpreterConfigurationSettings> interpreter();

}

File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/shell_script/ShellScriptConfigurationSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.shell_script

import com.github.alexandrecarlton.idea.settings.layout.configurations.ConfigurationSettings
import com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch.BeforeLaunchConfigurationSettings
import java.nio.file.Path

data class ShellScriptConfigurationSettings (
    override val name: String,

    override val shareThroughVcs: Boolean? = null,

    override val beforeLaunch: List<BeforeLaunchConfigurationSettings>? = null,

    val scriptPath: Path,

    val scriptOptions: String? = null,

    val interpreter: InterpreterConfigurationSettings? = null
) : ConfigurationSettings
